### PR TITLE
community/nodejs-current: upgrade to 11.1.0

### PR DESCRIPTION
**Pull request by grimd34th**

### CODE REVIEW

**Comment by TBK** [comment]
dont-run-gyp-files-for-bundled-deps.patch needs to be updated/checked causing CI to fail.

**Comment by TBK** [comment]
Node.js 10.0.0 needs a LibreSSL patch:
```
../src/node_crypto.cc:112:23: error: template-id 'GetSessionCallback<>' for 'SSL_SESSION* node::crypto::SSLWrap<node::TLSWrap>::GetSessionCallback(SSL*, const unsigned char*, int, int*)' does not match any template declaration
 template SSL_SESSION* SSLWrap<TLSWrap>::GetSessionCallback(
                       ^~~~~~~~~~~~~~~~
In file included from ../src/node_crypto.cc:26:0:
../src/node_crypto.h:256:23: note: candidate is: static SSL_SESSION* node::crypto::SSLWrap<Base>::GetSessionCallback(SSL*, unsigned char*, int, int*) [with Base = node::TLSWrap; SSL_SESSION = ssl_session_st; SSL = ssl_st]
   static SSL_SESSION* GetSessionCallback(SSL* s,
                       ^~~~~~~~~~~~~~~~~~
../src/node_crypto.cc: In static member function 'static void node::crypto::SecureContext::Init(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:372:41: error: 'TLS_method' was not declared in this scope
   const SSL_METHOD* method = TLS_method();
                                         ^
../src/node_crypto.cc:396:34: error: 'TLS_server_method' was not declared in this scope
       method = TLS_server_method();
                                  ^
../src/node_crypto.cc:398:34: error: 'TLS_client_method' was not declared in this scope
       method = TLS_client_method();
                                  ^
../src/node_crypto.cc:406:34: error: 'TLS_server_method' was not declared in this scope
       method = TLS_server_method();
                                  ^
../src/node_crypto.cc:410:34: error: 'TLS_client_method' was not declared in this scope
       method = TLS_client_method();
                                  ^
../src/node_crypto.cc:418:34: error: 'TLS_server_method' was not declared in this scope
       method = TLS_server_method();
                                  ^
../src/node_crypto.cc:422:34: error: 'TLS_client_method' was not declared in this scope
       method = TLS_client_method();
                                  ^
../src/node_crypto.cc:430:34: error: 'TLS_server_method' was not declared in this scope
       method = TLS_server_method();
                                  ^
../src/node_crypto.cc:434:34: error: 'TLS_client_method' was not declared in this scope
       method = TLS_client_method();
                                  ^
../src/node_crypto.cc:456:54: error: 'SSL_CTX_set_min_proto_version' was not declared in this scope
   SSL_CTX_set_min_proto_version(sc->ctx_, min_version);
                                                      ^
../src/node_crypto.cc:457:54: error: 'SSL_CTX_set_max_proto_version' was not declared in this scope
   SSL_CTX_set_max_proto_version(sc->ctx_, max_version);
                                                      ^
../src/node_crypto.cc:461:22: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_name_'
   if (RAND_bytes(sc->ticket_key_name_, sizeof(sc->ticket_key_name_)) <= 0 ||
                      ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:461:51: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_name_'
   if (RAND_bytes(sc->ticket_key_name_, sizeof(sc->ticket_key_name_)) <= 0 ||
                                                   ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:462:22: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_hmac_'
       RAND_bytes(sc->ticket_key_hmac_, sizeof(sc->ticket_key_hmac_)) <= 0 ||
                      ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:462:51: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_hmac_'
       RAND_bytes(sc->ticket_key_hmac_, sizeof(sc->ticket_key_hmac_)) <= 0 ||
                                                   ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:463:22: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_aes_'
       RAND_bytes(sc->ticket_key_aes_, sizeof(sc->ticket_key_aes_)) <= 0) {
                      ^~~~~~~~~~~~~~~
../src/node_crypto.cc:463:50: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_aes_'
       RAND_bytes(sc->ticket_key_aes_, sizeof(sc->ticket_key_aes_)) <= 0) {
                                                  ^~~~~~~~~~~~~~~
In file included from /usr/include/openssl/ssl.h:1702:0,
                 from ../src/node_crypto.h:39,
                 from ../src/node_crypto.cc:26:
../src/node_crypto.cc:467:36: error: 'TicketCompatibilityCallback' is not a member of 'node::crypto::SecureContext'
                                    SecureContext::TicketCompatibilityCallback);
                                    ^
../src/node_crypto.cc: In function 'X509_STORE* node::crypto::NewRootCertStore()':
../src/node_crypto.cc:752:23: error: 'X509_up_ref' was not declared in this scope
       X509_up_ref(cert);
                       ^
../src/node_crypto.cc: In static member function 'static void node::crypto::SecureContext::AddRootCerts(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:891:36: error: 'X509_STORE_up_ref' was not declared in this scope
   X509_STORE_up_ref(root_cert_store);
                                    ^
../src/node_crypto.cc: In static member function 'static void node::crypto::SecureContext::SetDHParam(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:956:39: error: 'DH_get0_pqg' was not declared in this scope
   DH_get0_pqg(dh, &p, nullptr, nullptr);
                                       ^
../src/node_crypto.cc: In static member function 'static void node::crypto::SecureContext::GetTicketKeys(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:1186:36: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_name_'
   memcpy(Buffer::Data(buff), wrap->ticket_key_name_, 16);
                                    ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:1187:41: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_hmac_'
   memcpy(Buffer::Data(buff) + 16, wrap->ticket_key_hmac_, 16);
                                         ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:1188:41: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_aes_'
   memcpy(Buffer::Data(buff) + 32, wrap->ticket_key_aes_, 16);
                                         ^~~~~~~~~~~~~~~
../src/node_crypto.cc: In static member function 'static void node::crypto::SecureContext::SetTicketKeys(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:1212:16: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_name_'
   memcpy(wrap->ticket_key_name_, Buffer::Data(args[0]), 16);
                ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:1213:16: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_hmac_'
   memcpy(wrap->ticket_key_hmac_, Buffer::Data(args[0]) + 16, 16);
                ^~~~~~~~~~~~~~~~
../src/node_crypto.cc:1214:16: error: 'class node::crypto::SecureContext' has no member named 'ticket_key_aes_'
   memcpy(wrap->ticket_key_aes_, Buffer::Data(args[0]) + 32, 16);
                ^~~~~~~~~~~~~~~
../src/node_crypto.cc: At global scope:
../src/node_crypto.cc:1322:55: error: no 'int node::crypto::SecureContext::TicketCompatibilityCallback(SSL*, unsigned char*, unsigned char*, EVP_CIPHER_CTX*, HMAC_CTX*, int)' member function declared in class 'node::crypto::SecureContext'
                                                int enc) {
                                                       ^
../src/node_crypto.cc:1430:14: error: prototype for 'SSL_SESSION* node::crypto::SSLWrap<Base>::GetSessionCallback(SSL*, const unsigned char*, int, int*)' does not match any in class 'node::crypto::SSLWrap<Base>'
 SSL_SESSION* SSLWrap<Base>::GetSessionCallback(SSL* s,
              ^~~~~~~~~~~~~
In file included from ../src/node_crypto.cc:26:0:
../src/node_crypto.h:256:23: error: candidate is: static SSL_SESSION* node::crypto::SSLWrap<Base>::GetSessionCallback(SSL*, unsigned char*, int, int*)
   static SSL_SESSION* GetSessionCallback(SSL* s,
                       ^~~~~~~~~~~~~~~~~~
../src/node_crypto.cc: In function 'v8::Local<v8::Object> node::crypto::X509ToObject(node::Environment*, X509*)':
../src/node_crypto.cc:1639:38: error: 'RSA_get0_key' was not declared in this scope
     RSA_get0_key(rsa, &n, &e, nullptr);
                                      ^
../src/node_crypto.cc: In static member function 'static void node::crypto::SSLWrap<Base>::GetTLSTicket(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:2050:49: error: there are no arguments to 'SSL_SESSION_get0_ticket' that depend on a template parameter, so a declaration of 'SSL_SESSION_get0_ticket' must be available [-fpermissive]
   SSL_SESSION_get0_ticket(sess, &ticket, &length);
                                                 ^
../src/node_crypto.cc:2050:49: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
../src/node_crypto.cc: In static member function 'static int node::crypto::SSLWrap<Base>::SSLCertCallback(SSL*, void*)':
../src/node_crypto.cc:2441:38: error: there are no arguments to 'SSL_get_tlsext_status_type' that depend on a template parameter, so a declaration of 'SSL_get_tlsext_status_type' must be available [-fpermissive]
   ocsp = SSL_get_tlsext_status_type(s) == TLSEXT_STATUSTYPE_ocsp;
                                      ^
../src/node_crypto.cc: In destructor 'virtual node::crypto::Hmac::~Hmac()':
../src/node_crypto.cc:3151:21: error: 'HMAC_CTX_free' was not declared in this scope
   HMAC_CTX_free(ctx_);
                     ^
../src/node_crypto.cc: In member function 'void node::crypto::Hmac::HmacInit(const char*, const char*, int)':
../src/node_crypto.cc:3184:23: error: 'HMAC_CTX_new' was not declared in this scope
   ctx_ = HMAC_CTX_new();
                       ^
../src/node_crypto.cc:3187:23: error: 'HMAC_CTX_free' was not declared in this scope
     HMAC_CTX_free(ctx_);
                       ^
../src/node_crypto.cc: In static member function 'static void node::crypto::Hmac::HmacDigest(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:3256:29: error: 'HMAC_CTX_free' was not declared in this scope
     HMAC_CTX_free(hmac->ctx_);
                             ^
../src/node_crypto.cc: In destructor 'virtual node::crypto::Hash::~Hash()':
../src/node_crypto.cc:3277:25: error: 'EVP_MD_CTX_free' was not declared in this scope
   EVP_MD_CTX_free(mdctx_);
                         ^
../src/node_crypto.cc: In member function 'bool node::crypto::Hash::HashInit(const char*)':
../src/node_crypto.cc:3310:27: error: 'EVP_MD_CTX_new' was not declared in this scope
   mdctx_ = EVP_MD_CTX_new();
                           ^
../src/node_crypto.cc:3313:27: error: 'EVP_MD_CTX_free' was not declared in this scope
     EVP_MD_CTX_free(mdctx_);
                           ^
../src/node_crypto.cc: In destructor 'virtual node::crypto::SignBase::~SignBase()':
../src/node_crypto.cc:3389:25: error: 'EVP_MD_CTX_free' was not declared in this scope
   EVP_MD_CTX_free(mdctx_);
                         ^
../src/node_crypto.cc: In member function 'node::crypto::SignBase::Error node::crypto::SignBase::Init(const char*)':
../src/node_crypto.cc:3405:27: error: 'EVP_MD_CTX_new' was not declared in this scope
   mdctx_ = EVP_MD_CTX_new();
                           ^
../src/node_crypto.cc:3408:27: error: 'EVP_MD_CTX_free' was not declared in this scope
     EVP_MD_CTX_free(mdctx_);
                           ^
../src/node_crypto.cc: In member function 'node::crypto::SignBase::Error node::crypto::Sign::SignFinal(const char*, int, const char*, unsigned char*, unsigned int*, int, int)':
../src/node_crypto.cc:3612:25: error: 'EVP_MD_CTX_free' was not declared in this scope
   EVP_MD_CTX_free(mdctx_);
                         ^
../src/node_crypto.cc: In member function 'node::crypto::SignBase::Error node::crypto::Verify::VerifyFinal(const char*, int, const char*, int, int, int, bool*)':
../src/node_crypto.cc:3793:25: error: 'EVP_MD_CTX_free' was not declared in this scope
   EVP_MD_CTX_free(mdctx_);
                         ^
../src/node_crypto.cc: In member function 'bool node::crypto::DiffieHellman::Init(const char*, int, int)':
../src/node_crypto.cc:4053:43: error: 'DH_set0_pqg' was not declared in this scope
       !DH_set0_pqg(dh, bn_p, nullptr, bn_g)) {
                                           ^
../src/node_crypto.cc: In member function 'bool node::crypto::DiffieHellman::Init(const char*, int, const char*, int)':
../src/node_crypto.cc:4070:43: error: 'DH_set0_pqg' was not declared in this scope
   if (!DH_set0_pqg(dh, bn_p, nullptr, bn_g)) {
                                           ^
../src/node_crypto.cc: In static member function 'static void node::crypto::DiffieHellman::GenerateKeys(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:4163:51: error: 'DH_get0_key' was not declared in this scope
   DH_get0_key(diffieHellman->dh, &pub_key, nullptr);
                                                   ^
../src/node_crypto.cc: In lambda function:
../src/node_crypto.cc:4192:41: error: 'DH_get0_pqg' was not declared in this scope
     DH_get0_pqg(dh, &p, nullptr, nullptr);
                                         ^
../src/node_crypto.cc: In lambda function:
../src/node_crypto.cc:4201:41: error: 'DH_get0_pqg' was not declared in this scope
     DH_get0_pqg(dh, nullptr, nullptr, &g);
                                         ^
../src/node_crypto.cc: In lambda function:
../src/node_crypto.cc:4210:38: error: 'DH_get0_key' was not declared in this scope
     DH_get0_key(dh, &pub_key, nullptr);
                                      ^
../src/node_crypto.cc: In lambda function:
../src/node_crypto.cc:4219:39: error: 'DH_get0_key' was not declared in this scope
     DH_get0_key(dh, nullptr, &priv_key);
                                       ^
../src/node_crypto.cc: In lambda function:
../src/node_crypto.cc:4328:71: error: 'DH_set0_key' was not declared in this scope
          [](DH* dh, BIGNUM* num) { return DH_set0_key(dh, num, nullptr); },
                                                                       ^
../src/node_crypto.cc: In static member function 'static void node::crypto::DiffieHellman::SetPublicKey(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:4329:22: error: invalid user-defined conversion from 'node::crypto::DiffieHellman::SetPublicKey(const v8::FunctionCallbackInfo<v8::Value>&)::<lambda(DH*, BIGNUM*)>' to 'int (*)(DH*, BIGNUM*) {aka int (*)(dh_st*, bignum_st*)}' [-fpermissive]
          "Public key");
                      ^
../src/node_crypto.cc:4328:32: note: candidate is: node::crypto::DiffieHellman::SetPublicKey(const v8::FunctionCallbackInfo<v8::Value>&)::<lambda(DH*, BIGNUM*)>::operator void (*)(DH*, BIGNUM*)() const <near match>
          [](DH* dh, BIGNUM* num) { return DH_set0_key(dh, num, nullptr); },
                                ^
../src/node_crypto.cc:4328:32: note:   no known conversion from 'void (*)(DH*, BIGNUM*) {aka void (*)(dh_st*, bignum_st*)}' to 'int (*)(DH*, BIGNUM*) {aka int (*)(dh_st*, bignum_st*)}'
../src/node_crypto.cc:4298:6: note:   initializing argument 2 of 'static void node::crypto::DiffieHellman::SetKey(const v8::FunctionCallbackInfo<v8::Value>&, int (*)(DH*, BIGNUM*), const char*)'
 void DiffieHellman::SetKey(const v8::FunctionCallbackInfo<v8::Value>& args,
      ^~~~~~~~~~~~~
../src/node_crypto.cc: In lambda function:
../src/node_crypto.cc:4340:71: error: 'DH_set0_key' was not declared in this scope
          [](DH* dh, BIGNUM* num) { return DH_set0_key(dh, nullptr, num); },
                                                                       ^
../src/node_crypto.cc: In static member function 'static void node::crypto::DiffieHellman::SetPrivateKey(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:4341:23: error: invalid user-defined conversion from 'node::crypto::DiffieHellman::SetPrivateKey(const v8::FunctionCallbackInfo<v8::Value>&)::<lambda(DH*, BIGNUM*)>' to 'int (*)(DH*, BIGNUM*) {aka int (*)(dh_st*, bignum_st*)}' [-fpermissive]
          "Private key");
                       ^
../src/node_crypto.cc:4340:32: note: candidate is: node::crypto::DiffieHellman::SetPrivateKey(const v8::FunctionCallbackInfo<v8::Value>&)::<lambda(DH*, BIGNUM*)>::operator void (*)(DH*, BIGNUM*)() const <near match>
          [](DH* dh, BIGNUM* num) { return DH_set0_key(dh, nullptr, num); },
                                ^
../src/node_crypto.cc:4340:32: note:   no known conversion from 'void (*)(DH*, BIGNUM*) {aka void (*)(dh_st*, bignum_st*)}' to 'int (*)(DH*, BIGNUM*) {aka int (*)(dh_st*, bignum_st*)}'
../src/node_crypto.cc:4298:6: note:   initializing argument 2 of 'static void node::crypto::DiffieHellman::SetKey(const v8::FunctionCallbackInfo<v8::Value>&, int (*)(DH*, BIGNUM*), const char*)'
 void DiffieHellman::SetKey(const v8::FunctionCallbackInfo<v8::Value>& args,
      ^~~~~~~~~~~~~
../src/node_crypto.cc: In function 'void node::crypto::GetSSLCiphers(const v8::FunctionCallbackInfo<v8::Value>&)':
../src/node_crypto.cc:5059:41: error: 'TLS_method' was not declared in this scope
   SSL_CTX* ctx = SSL_CTX_new(TLS_method());
                                         ^
../src/node_crypto.cc: In instantiation of 'static int node::crypto::SSLWrap<Base>::SSLCertCallback(SSL*, void*) [with Base = node::TLSWrap; SSL = ssl_st]':
../src/node_crypto.cc:128:65:   required from here
../src/node_crypto.cc:2441:36: error: 'SSL_get_tlsext_status_type' was not declared in this scope
   ocsp = SSL_get_tlsext_status_type(s) == TLSEXT_STATUSTYPE_ocsp;
          ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~
../src/node_crypto.cc: In instantiation of 'static void node::crypto::SSLWrap<Base>::GetTLSTicket(const v8::FunctionCallbackInfo<v8::Value>&) [with Base = node::TLSWrap]':
../src/node_crypto.cc:1403:3:   required from 'static void node::crypto::SSLWrap<Base>::AddMethods(node::Environment*, v8::Local<v8::FunctionTemplate>) [with Base = node::TLSWrap]'
../src/node_crypto.cc:108:69:   required from here
../src/node_crypto.cc:2050:26: error: 'SSL_SESSION_get0_ticket' was not declared in this scope
   SSL_SESSION_get0_ticket(sess, &ticket, &length);
   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by andypost** [comment]
10.1 is out https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V10.md#10.1.0
Also it needs rebase to fix merge conflicts

**Comment by jirutka** [comment]
@ncopa reminder

**Comment by f3l1x** [comment]
Hi gyus. Any way to how help you with that? I would like to install node 10. :-)

File: community/nodejs-current/APKBUILD [hunk 1]
```diff
@@ -12,7 +12,7 @@
 #
 pkgname=nodejs-current
 # The current stable version, i.e. non-LTS.
-pkgver=9.11.1
+pkgver=10.0.0
```
Comment (andypost) [inline community/nodejs-current/APKBUILD]: Current is 10.7.0

Comment (andypost) [reply community/nodejs-current/APKBUILD]: current is 10.10.0 now

File: community/nodejs-current/APKBUILD [hunk 2]
```diff
@@ -1,27 +1,20 @@
...
+
 pkgdesc="JavaScript runtime built on V8 engine - current stable version"
 url="https://nodejs.org/"
 arch="all"
 license="MIT"
 depends="ca-certificates"
 depends_dev="libuv"
 # gold is needed for mksnapshot
-makedepends="$depends_dev python2 openssl1.0-dev zlib-dev libuv-dev linux-headers
+makedepends="$depends_dev python2 openssl1.1-dev zlib-dev libuv-dev linux-headers
```
Comment (andypost) [inline community/nodejs-current/APKBUILD]: `openssl1.1-dev` is in testing repo and could not be used in community

**Comment by andypost** [comment]
nodejs-current is 11.1.0 now and 10.13.0 is LTS

**Comment by andypost** [comment]
Filed upgrade for LTS https://github.com/alpinelinux/aports/pull/5558

File: community/nodejs-current/APKBUILD (hunk 2 above)
Comment (timbru31) [reply community/nodejs-current/APKBUILD]: as in #5558 the correct package is openssl-dev

File: community/nodejs-current/APKBUILD [hunk 3]
```diff
@@ -1,27 +1,20 @@
...
-#     - CVE-2017-15896
-#     - CVE-2017-15897
-#
 pkgname=nodejs-current
 # The current stable version, i.e. non-LTS.
-pkgver=9.11.1
-pkgrel=4
+
+pkgver=10.10.0
+pkgrel=1
```
Comment (timbru31) [inline community/nodejs-current/APKBUILD]: Node.js v11.1.0 is already out, can you update this PR? AFAIK pgrel should be set to 0

File: community/nodejs-current/dont-run-gyp-files-for-bundled-deps.patch [hunk 4]
```diff
@@ -4,18 +4,18 @@ Subject: Disable running gyp files for bundled deps
 
 Author: Stephen Gallagher <sgallagh@redhat.com>
 
-Modified 2016-11-26 by Jakub Jirutka <jakub@jirutka.cz> to update for
-Node.js 7.2.0
+Modified 2016-11-26 by Jakub Jirutka <jakub@jirutka.cz> to update for Node.js 7.2.0
+Modified 2018-04-27 to Node.js 10.0.0
```
Comment (timbru31) [inline community/nodejs-current/dont-run-gyp-files-for-bundled-deps.patch]: This should be changed to 11.1.0

File: community/nodejs-current/APKBUILD [hunk 5]
```diff
@@ -1,27 +1,20 @@
 # Contributor: Jakub Jirutka <jakub@jirutka.cz>
 # Maintainer: Jose-Luis Rivas <ghostbar@riseup.net>
 #
-# secfixes:
```
Comment (timbru31) [inline community/nodejs-current/APKBUILD]: @jirutka said that secfixes should remain for history purposes

File: community/nodejs-current/APKBUILD (hunk 2 above)
Comment (ncopa) [reply community/nodejs-current/APKBUILD]: `openssl1.1-dev` has been renamed to `openssl-dev` and moved to main.

**Comment by ncopa** [comment]
this is fixed with d103c6e96b8aa8a7fe28db2f48b8e7ff0e4cd686

**Pull request closed without merging**